service/frontend/workflowHandler.go

```diff
@@ -160,6 +160,13 @@ func NewWorkflowHandler(
 				return float64(config.RPS())
 			},
 			func(domain string) float64 {
+				if monitor := resource.GetMembershipMonitor(); monitor != nil {
+					ringSize, err := monitor.GetMemberCountWithRole(common.FrontendServiceName)
+					if err == nil && ringSize > 0 {
```

Comment:
Is this always the case? If true we should consider removing the domainRPS logic completely.